src/shogun/clustering/KMeansBase.cpp

```diff
@@ -0,0 +1,381 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 1999-2008 Gunnar Raetsch
+ * Written (W) 2007-2009 Soeren Sonnenburg
+ * Copyright (C) 1999-2009 Fraunhofer Institute FIRST and Max-Planck-Society
+ */
+
+#include <shogun/clustering/KMeansBase.h>
+#include <shogun/distance/Distance.h>
+#include <shogun/distance/EuclideanDistance.h>
+#include <shogun/labels/Labels.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/mathematics/Math.h>
+#include <shogun/base/Parallel.h>
+#include <shogun/mathematics/eigen3.h>
+
+#ifdef HAVE_PTHREAD
```

Comment:
i know that this is just the code moved from the original implementation, but what's the use of including pthread? couldn't find any pthread calls...